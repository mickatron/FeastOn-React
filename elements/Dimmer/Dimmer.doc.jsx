/* eslint-disable */
import React from 'react';
import Component from './Dimmer';
import PropTypes from 'prop-types';
import asType from '../../proptypes/as';
import classNameType from '../../proptypes/className';

/**
 * A Dimmer component used most commonly as a modal background.
 * 
 * ** Sub Components **
 * * `DimmerHeader` as `Dimmer.Header`
 * * `DimmerContent` as `Dimmer.Content` 
 * * `DimmerFooter` as `Dimmer.Footer`
 *
 * @composedWith UiWrapper
*/
function Dimmer(props) {
  return <Component {...props} />;
}

export default Dimmer;
