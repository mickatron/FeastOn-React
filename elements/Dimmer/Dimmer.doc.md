
### Default Dimmer.
```js
<div style={{position:'relative', width: '100%', height: '200px'}}>
    <span>Proin id ultrices nunc. Cras leo orci, ultrices auctor lacinia sed, maximus a diam. Quisque convallis posuere est, vitae venenatis urna interdum eu.</span>
<Dimmer>
    <span>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam quis urna ac eros hendrerit convallis. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.</span>
</Dimmer>
</div>
```