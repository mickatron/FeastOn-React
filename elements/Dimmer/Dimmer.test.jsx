import { shallow } from 'enzyme';
import React from 'react';
import Dimmer from './Dimmer';

describe('Dimmer', () => {
  let component = null;

  it('renders', () => {
    component = shallow((
      <Dimmer
        className="test"
        modifier={['outline']}
      >
        <p>Test Component Content.</p>
      </Dimmer>
    ));
    expect(component).toMatchSnapshot();
  });
});
