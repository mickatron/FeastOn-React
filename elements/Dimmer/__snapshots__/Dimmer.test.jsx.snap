// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dimmer renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UiWrapper
    as="div"
    className="test"
    modifier={
      Array [
        "outline",
      ]
    }
    widthClass={undefined}
  >
    <p>
      Test Component Content.
    </p>
  </UiWrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p>
        Test Component Content.
      </p>,
      "className": "test Dimmer Dimmer--outline",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test Component Content.",
      },
      "ref": null,
      "rendered": "Test Component Content.",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p>
          Test Component Content.
        </p>,
        "className": "test Dimmer Dimmer--outline",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test Component Content.",
        },
        "ref": null,
        "rendered": "Test Component Content.",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
